### ISSUE TITLE
C# compiler should treat return value of SingleOrDefault as nullable for enumerables with reference types

### ISSUE THREAD

**Issue by oliverhanappi** [dotnet/roslyn#38941]
**Version Used**: .NET Core SDK 3.0.100

**Steps to Reproduce**:

```csharp
#nullable enable

public class DemoClass
{
    public object DemoMethod() => new object[0].SingleOrDefault();
}
```

**Expected Behavior**:

C# compiler detects the return value of SingleOrDefault as nullable and therefore emits warning CS8603.

**Actual Behavior**:

No warning is emitted.

**Comment by huoyaoyuan** [comment]
`System.Linq` is nullable annotated in **master** branch of corefx, but not in **release/3.0**. So there's nothing unexpected in compiler.
The compiler should provide some diagnostics showing that you are using nullable-oblivious stuff.

**Comment by benallred** [comment]
Running into this myself. I understand that we're going to have to wait for libraries (I wouldn't normally lump `System.Linq` under "libraries" but that applies here, I guess) to support the annotations. In the meantime, is there something one can do to generate warnings on nullable-oblivious dependencies?

**Comment by danstur** [comment]
Is this planned to be fixed for the 5.0 release? Having one of the cornerstone libraries in .NET not be aware of nullability severely limits the usefulness of this feature.

**Comment by jcouv** [comment]
It looks like this was fixed. Code below now produces a warning.

```
#nullable enable
using System.Linq;
public class DemoClass
{
    public object DemoMethod() => new object[0].SingleOrDefault(); // warning CS8603: Possible null reference return.
}
```
([sharplab](https://sharplab.io/#v2:CYLg1APgxAdgrgGwQQwEYIKYAIMzZgWACgABABixIEYA6AGQEsYBHAbmJIGZKAmLAEQwBbAPYBhFAGdJxAN7EsiytxGoAVhgDGAFwHCRAWQzaAFiOAAKAJRYAvAD4sMDAHcsqjToDaZALo0AZSYAc0wAeQAnQQAzZERta1YsAHpkrBdkCJgQrDEAgA4ANjJOECwABRFpBnRseCQsCIxojCaYTWwSAHYaYgBfIA==))

I'll go ahead and close this. Thanks
